racecar_bringup/launch/gro525.launch.py

```diff
@@ -1,41 +1,68 @@
 from launch import LaunchDescription
-from launch.actions import DeclareLaunchArgument, SetLaunchConfiguration
 from launch_ros.actions import Node
-from launch.actions import IncludeLaunchDescription
-from launch_xml.launch_description_sources import XMLLaunchDescriptionSource
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
 from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch_xml.launch_description_sources import XMLLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration
 from ament_index_python.packages import get_package_share_directory
 
-def generate_launch_description():
-    # Get the package directories
-    rosbridge_server_pkg_share_dir = get_package_share_directory('rosbridge_server')
-    web_video_server_pkg_share_dir = get_package_share_directory('web_video_server')
-    racecar_web_interface_pkg_share_dir = get_package_share_directory('racecar_web_interface')
-    racecar_bringup_pkg_share_dir = get_package_share_directory('racecar_bringup')
-    
-    port_argument = DeclareLaunchArgument(
-        'port', default_value='9090',
-        description='Port for rosbridge websocket'
-    )
-    
 
-    # Launch rosbridge_websocket_launch.xml from rosbridge_server package
-    rosbridge_launch = IncludeLaunchDescription(
-        XMLLaunchDescriptionSource([rosbridge_server_pkg_share_dir, '/launch/rosbridge_websocket_launch.xml']),
-        launch_arguments={'port': '9090'}.items()
-    )
+racecar_bringup_dir = get_package_share_directory("racecar_bringup")
+rosbridge_server_dir = get_package_share_directory("rosbridge_server")
 
-    #Launch web_video_server node from web_video_server package
-    web_video_server_node = Node(
-        package='web_video_server',
-       executable='web_video_server',
-        name='web_video_server',
-        output='screen',
-        parameters=[{'address': '10.42.0.1'}]  # Set the address parameter to "10.42.0.1"
-    )
+rosbridge_port_arg = DeclareLaunchArgument(
+    "port", default_value="9090", description="Port for rosbridge websocket"
+)
+
+host_address_arg = DeclareLaunchArgument(
+    "host_address", default_value="127.0.0.1", description="Address of the RaspberryPi"

```

Comment:
I would keep the 10.42.0.1 for simplicity since it will be easier for student to not worry about the IP address